lib/puppet/util/windows/registry.rb

```diff
@@ -220,6 +220,16 @@ def reg_query_info_key_max_lengths(key)
     def read(key, name_ptr, *rtype)
       result = nil
 
+      string_sanitizer = lambda do |value|
+        return value unless value.respond_to?(:gsub)
+
+        # Replace byte sequences that do not compose
+        # to valid UTF8, and null bytes with the Unicode
+        # Replacement Character: "�"
+        return value.scrub!.gsub(/\0/, "\uFFFD") if value.respond_to?(:scrub)
+        value.gsub(/\0/, "\uFFFD") # Ruby < 2.1 does not implement scrub
```

Comment:
We only support Ruby 2.4+ on Puppet 5.5:
https://puppet.com/docs/puppet/5.5/system_requirements.html